cmd/interchaincmd/relayercmd/deploy.go

```diff
@@ -371,23 +377,48 @@ func CallDeploy(_ []string, flags DeployFlags, network models.Network) error {
 						return err
 					}
 				}
+				balance, err := evm.GetPrivateKeyBalance(client, privateKey)
+				if err != nil {
+					return err
+				}
+				if balance.Cmp(big.NewInt(0)) == 0 {
+					return fmt.Errorf("destination %s funding key as no balance", destination.blockchainDesc)
+				}
+				balanceBigFlt := new(big.Float).SetInt(balance)
+				balanceBigFlt = balanceBigFlt.Quo(balanceBigFlt, new(big.Float).SetInt(vm.OneAvax))
+				balanceFlt, _ := balanceBigFlt.Float64()
+				balanceFlt -= aproxFundingFee
 				var amountFlt float64
-				if flags.Amount != 0 {
+				switch {
+				case !isCChainDestination && flags.Amount != 0:
 					amountFlt = flags.Amount
-				} else {
+				case isCChainDestination && flags.CChainAmount != 0:
+					amountFlt = flags.CChainAmount
+				default:
 					amountFlt, err = app.Prompt.CaptureFloat(
-						"Amount to transfer",
+						fmt.Sprintf("Amount to transfer (available: %f)", balanceFlt),
```

Comment:
debug or you want to use ux.PrintToUser?

Reply:
this is just a formatting directive